docs/machine-learning/how-to-guides/matchup-app-infer-net.md

```diff
@@ -0,0 +1,158 @@
+
+## What is probabilistic programming? 
+
+Probabilistic programming allows us to create statistical models of real-world processes. 
+
+## Prerequisites
+
+- Local development environment setup
+
+  This how-to guide expects you to have a machine you can use for development. The .NET topic [Get Started in 10 minutes](https://www.microsoft.com/net/core) has instructions for setting up your local development environment on Mac, PC, or Linux.
+
+## Create your app
+
+1. Open a new command prompt and run the following commands:
+
+```console
+dotnet new console -o myApp
+cd myApp
+```
+
+The `dotnet` command creates a `new` application of type `console`. The `-o` parameter creates a directory named `myApp` where your app is stored, and populates it with the required files. The `cd myApp` command puts you into the newly created app directory.
+
+## Install Infer.NET package
+
+To use Infer.NET, you need to install the Microsoft.ML.Probabilistic package. In your command prompt, run the following command:
+
+```console
+dotnet add package Microsoft.ML.Probabilistic.Compiler
+```
+
+## Design your model
+
+We'll use for our example sample table tennis or foosball matches played in the office. We have the participants and outcome of each match.  We want to infer the player's skills from this data. We’ll assume that each player has a normally distributed latent skill and their performance in a given match is a noisy version of this skill. The data constrains the winner’s performance to be greater than the loser’s performance. This is a simplified version of the popular [TrueSkill](https://www.microsoft.com/en-us/research/project/trueskill-ranking-system/) model, which also supports teams, draws, and other extensions. An [advanced version](https://www.microsoft.com/en-us/research/publication/trueskill-2-improved-bayesian-skill-rating-system/) of this model is used for matchmaking in the best-selling game titles Halo and Gears of War.
+
+We need to list the inferred player skills, alongside with their variance – the measure of uncertainty around the skills.
+
+*Game result sample data*
+
+Game |Winner | Loser
+---------|----------|---------
+ 1 | Player 0 | Player 1
+ 2 | Player 0 | Player 3
+ 3 | Player 0 | Player 4
+ 4 | Player 1 | Player 2
+ 5 | Player 3 | Player 1
+ 6 | Player 4 | Player 2
+
+With a closer look at the sample data, you’ll notice that players 3 and 4 both have one win and one loss. Let's see what the rankings look like using probablistic programming.
+
+## Write some code
+
+Having designed the model, it’s time to express it as a probabilistic program using the Infer.NET modeling API. Open `Program.cs` in your favorite text editor and replace all of the code with the following:
```

Comment:
replace all of the code with the following -> replace all of its contents with the following code